pymc/sampling/mcmc.py

```diff
@@ -1285,14 +1293,33 @@ def _iter_sample(
     step.set_rng(rng)
 
     point = start
+    initial_draw_idx = 0
+    step.tune = bool(tune)
+    if hasattr(step, "reset_tuning"):
+        step.reset_tuning()
     if isinstance(trace, ZarrChain):
         trace.link_stepper(step)
+        stored_draw_idx = trace._sampling_state.draw_idx[chain]
```

Comment:
Same here all this logic including the old `link_stepper` can have a sensible default in the base trace class so you don't need to worry about what kind of trace you have here. Just make `link_stepper` a no op and `stored_draw_idx` to be zero by default?